zerver/lib/export.py

```diff
@@ -1147,8 +1153,10 @@ def custom_fetch_scheduled_messages(response: TableData, context: Context) -> No
     realm = context["realm"]
     exportable_scheduled_message_ids = context["exportable_scheduled_message_ids"]
 
-    query = ScheduledMessage.objects.filter(realm=realm, id__in=exportable_scheduled_message_ids)
-    rows = make_raw(list(query))
+    query = ScheduledMessage.objects.filter(
+        realm=realm, id__in=exportable_scheduled_message_ids
+    ).prefetch_related("attachment_set")
+    rows = make_raw(list(query), is_message_table=True)
```

Comment:
Isn't this data already present in the export, via `fetch_attachment_data`?

Reply:
Yes, it is present, and as a result in `import_attachments()` each attachment has direct access to its related  messages.

But the problem is `import_attachments()` is called after `import_message_data` (which calls `fix_upload_links`)
and for some reason this [comment](https://github.com/zulip/zulip/blob/main/zerver/lib/import_realm.py#L1437) says that we should respect that order.

I will try a workaround which is simply keep that order as it is, but before we import the messages we read the attachments file (won't import them yet) and exclusively do that mapping.